src/pydvl/value/shapley/montecarlo.py

```diff
@@ -0,0 +1,521 @@
+        :attr:`~pydvl.utils.dataset.Dataset.indices`
+    :param config: Object configuring parallel computation, with cluster
+        address, number of cpus, etc.
+    :param progress: Whether to display progress bars for each job.
+    :return: Tuple with the first element being an ordered Dict of approximate
+        Shapley values for the indices, the second being their standard error
+    """
+    parallel_backend = init_parallel_backend(config)
+    u_id = parallel_backend.put(u)
+
+    # FIXME? max_iterations has different semantics in permutation-based methods
+    map_reduce_job: MapReduceJob["NDArray", MonteCarloResults] = MapReduceJob(
+        map_func=_combinatorial_montecarlo_shapley,
+        reduce_func=disjoint_reducer,
+        map_kwargs=dict(u=u_id, max_iterations=max_iterations, progress=progress),
+        chunkify_inputs=True,
+        n_jobs=n_jobs,
+        config=config,
+    )
+    results = map_reduce_job(u.data.indices)[0]
+    sorted_shapley_values = sort_values(
+        {u.data.data_names[i]: v for i, v in enumerate(results.values)}
+    )
+    montecarlo_errors = {u.data.data_names[i]: v for i, v in enumerate(results.stderr)}
+
+    return sorted_shapley_values, montecarlo_errors
+
+
+class OwenAlgorithm(Enum):
+    Standard = "standard"
+    Antithetic = "antithetic"
+
+
+def _owen_sampling_shapley(
+    indices: Sequence[int],
+    u: Utility,
+    method: OwenAlgorithm,
+    max_iterations: int,
+    max_q: int,
+    *,
+    progress: bool = False,
+    job_id: int = 1,
+) -> MonteCarloResults:
+    r"""This is the algorithm as detailed in the paper: to compute the outer
+    integral over q ∈ [0,1], use uniformly distributed points for evaluation
+    of the integrand. For the integrand (the expected marginal utility over the
+    power set), use Monte Carlo.
+
+    .. todo::
+        We might want to try better quadrature rules like Gauss or Rombert or
+        use Monte Carlo for the double integral.
+
+    :param indices: Indices to compute the value for
+    :param u: Utility object with model, data, and scoring function
+    :param method: Either :attr:`~OwenAlgorithm.Full` for $q \in [0,1]$ or
+        :attr:`~OwenAlgorithm.Halved` for $q \in [0,0.5]$ and correlated samples
+    :param max_iterations: Number of subsets to sample to estimate the integrand
+    :param max_q: number of subdivisions for the integration over $q$
+    :param progress: Whether to display progress bars for each job
+    :param job_id: For positioning of the progress bar
+    :return: Values and standard errors
+    """
+    values = np.zeros(len(u.data))
+
+    q_stop = {OwenAlgorithm.Standard: 1.0, OwenAlgorithm.Antithetic: 0.5}
+    q_steps = np.linspace(start=0, stop=q_stop[method], num=max_q)
+
+    index_set = set(indices)
+    for i in maybe_progress(indices, progress, position=job_id):
+        e = np.zeros(max_q)
+        subset = np.array(list(index_set.difference({i})))
```

Comment:
I actually went and timed this against using numpy's setxor1d and it turns there isn't much difference.
setxor1d scales better but you need a lot of indices for it to even matter.

Reply:
If "a lot" is millions, and the difference is in ms, then we can totally ignore this